src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -537,6 +537,14 @@ export type MarkdownEditorProps = {
   usersForMentions: UserItem[];
 };
 
+/**
+ * Replace all mentions wrapped in the special `<a data-hyp-mention />`tag with
+ * their corresponding plain-text representation
+ */
+function unwrapMentions(text: string) {
+  return text.replace(/<a[^>]*data-hyp-mention[^>]*>([^<]*)<\/a>/g, '$1');
```

Comment:
Right now this discards the userid for every mention, as it assumes the tag content is the username, with which the userid can be composed back from.

In future, when mentioning via display name is supported, it may be a good idea to track/save the userids.

Reply:
Edit: The previous comment is wrong, as `<` characters _should_ be escaped. We will need to gracefully handle the case where a malformed mention appears in the text though.

Reply:
I think using DOM APIs as you suggested in https://github.com/hypothesis/client/pull/6787#discussion_r1941031848 is definitely the way to go, and would solve this problem.